### DISCUSSION TITLE
Convert DV profile 7 to DV profile 5

### DISCUSSION THREAD

**Discussion by imtiendat0311** [quietvoid/dovi_tool#176] [Q&A] [answered]
Hi quiet void,
Im new to this. I have a remux file that dv profile 7. But in order to playback DV in macOS I need to convert it to DV profile 5 . So is it possible to convert DV7 to DV5 profile using your tool. 
Thanks

**Comment by sw5163** [comment, marked as answer]
... Yes it's possible.
Just check the discussion below: [Dual layer FEL dolby vision to Apple compatible BL+RPU](https://github.com/quietvoid/dovi_tool/discussions/168)
It has been confirmed Apple supports BL+RPU playback.
See also https://forum.makemkv.com/forum/viewtopic.php?p=123899#p123899

But if you mean proper conversion from profile 7 HDR10 BL to profile 5 ICtCp with correct reshaping and RPU data, then no, there isn't a public way yet.

**Reply by imtiendat0311** [reply]
any noticeable different in color when using this method compare with play true profile 5 ?

**Reply by sw5163** [reply]
Profile 7 FEL is only supported on very few devices such as Bluray disc player like Sony x800m2.
It has 12 bit YCbCr 4:2:2 color sampling.

Apple doesn't support the extra 2 bits in EL, so BL+RPU (10 bit YCbCr 4:2:0 + dynamic tone mapping) is currently the best quality you can get on Apple from the P7 file you have.

Real Profile 5 use 10 bit ICtCp (I'm not sure if it's 4:2:0), it's a more efficient encoding and is equivalent to 11.5 bit YCbCr.
Very few Apple devices have real 10 bit panel, so it's extremely hard to tell the difference between color.
The bitrate however, is noticeable between streaming P5 and Bluray.

I would say the video quality from that method is better than streaming P5 and probably is the highest quality on Apple for now. It's how I prefer to watch contents.

Oh and avoid using third party player like Infuse, use the native player (quicktime I think).

**Reply by imtiendat0311** [reply]
is the mini led macbook come with true 10 bit panel ? Moreover, how to test it and check it
Thank you all for your respond

**Reply by sw5163** [reply]
Not sure, but don't worry about it, dithering algorithm nowadays is highly advanced. It's hard to tell a difference between 8 bit + dithering and 10 bit without dithering.

**Comment by imtiendat0311** [comment]
Also other note i see that @sw5163  success with trick apple software to playback dv for FEL dv 7 but is it possible to do with MEL since my remux file is MEL BL 4k and EL 1080p  
Thanks all

**Reply by sw5163** [reply]
What? No, I never succeeded in playing FEL.

FEL=BL+EL+RPU
MEL=BL+RPU
MEL means the 1080p track only has RPU.

My workflow discards the EL, so it is MEL playback.
Follow the procedure and you will be fine.
For the remux part 3, you can use mp4box if mp4muxer is too slow.
3. mp4box -add BL_RPU.hevc -new p5.dvh1.mp4
I personally use it instead of mp4muxer, mp4muxer is the official tool though.

mp4box also supports 7.1 lossless pcm, see [Apple BL+RPU and 7.1 lossless audio](https://forum.makemkv.com/forum/viewtopic.php?p=124598#p124598)
You could also help test if those files trigger Apple's spatial audio, thx. I don't have capable device.

**Reply by imtiendat0311** [reply]
~~for some reason when i use mp4box ( macos ) to mux i only get HDR format : SMPTE ST 2086, HDR10 compatible
only when i use mp4muxer by dolby  i can get HDR format  : Dolby Vision, Version 1.0, dvhe.05.06, BL+RPU / SMPTE ST 2086, HDR10 compatible.~~
update to mp4box version 2.1 fix the problem
thank you , i got it proper play and running but seem to my eye it's like hdr10 only. I will download mp4 streaming version and compare thank you

**Reply by sw5163** [reply]
> i got it proper play and running but seem to my eye it's like hdr10 only.

If the workflow is correct, and final video is P5 mp4 with dvh1 tag then it is DV with dynamic tonemapping. It's been proved on mkvforum.
If the file seems like hdr10, it could because your monitor is 1600nit and no content in the file exceeds that. In that case, DV and HDR10 may look the same.

**Reply by imtiendat0311** [reply]
> > i got it proper play and running but seem to my eye it's like hdr10 only.
> 
> If the workflow is correct, and final video is P5 mp4 with dvh1 tag then it is DV with dynamic tonemapping. It's been proved on mkvforum. If the file seems like hdr10, it could because your monitor is 1600nit and no content in the file exceeds that. In that case, DV and HDR10 may look the same.

this is my workflow 
i got mkv remux file with 1 video stream that include bl + el + rpu  and true hd audio stream
i demux the video stream to hevc and audio stream to mp4
```shell
ffmpeg -i filename.mkv -c copy -map 0:0 out.hevc -c copy -map 0:1 -c flac -strict -2 flac.mp4
```
using dovi_tool to demux the hevc to BL and EL  and mux back discard the EL layer with config file 
```shell
dovi_tool demux -i out.hevc
dovi_tool --edit-config ok.json mux --bl BL.hvec --el EL.hvec --discard
```
using mp4box to mux it to mp4
```shell
mp4box -add BL+RPU.hvec:dv-profile=5 -add flac.mp4:lang=eng -new out.mp4 
```
i got file HDR format : Dolby Vision, Version 1.0, dvhe.05.06, BL+RPU, HDR10 compatible / SMPTE ST 2086, HDR10 compatible / SMPTE ST 2086 - which is compatible with quicktime

also other note how can i import it to iphone ipad.
I tried add to tv app and sync it using itunes but my iphone seem skip the file.
Thanks

**Reply by sw5163** [reply]
The edit config(remove_mapping) is only needed when it's FEL and has reshaping data.
I would do something like:
```shell
ffmpeg -i filename.mkv -map 0:0 -c copy BL_EL_RPU.hevc -map 0:1 -c pcm_s32le pcm.mov
```
Using default mode 2 in dovi_tool because "remove_mapping" is not needed in MEL.
```shell
dovi_tool -m 2 convert BL_EL_RPU.hevc --discard
```
Using mp4box to mux it to mp4
```shell
mp4box -add BL_RPU.hevc -add pcm.mov:lang=eng -add sub.srt:lang=eng:hdlr=sbtl -new out.mp4 
```
mp4box use profile 5 dvh1 by default.
PCM is for 7.1 lossless, flac and alac can't play 7.1 on Apple.
lang=eng:hdlr=sbtl for subtitle to default show up.
> also other note how can i import it to iphone ipad. I tried add to tv app and sync it using itunes but my iphone seem skip the file. Thanks

I don't use macos, but I think on Apple's computer it won't be hard to do it. Finder? or air drop should be able to do it. Import the file to the File App on iOS should also be fine.
On windows I use SMB and share folder to let iOS device access contents through File App.

**Reply by imtiendat0311** [reply]
after testing with other file i could fairly say that it output correct dolby vision hdr. Thank you for your advise.
Also could you willing to guide me or provide me documentation about dovi tool and all stuff. I would like to learn more about this.

**Reply by sw5163** [reply]
Im not professional. Dolby has many documentation on their website.
https://professionalsupport.dolby.com/s/article/Dolby-Vision-Metadata-Levels?language=en_US
The readme of dovi_tool is already very detailed.
